server/routes/pages.js

```diff
@@ -11,6 +11,12 @@ const { getTemplates, sortDocs, stringTemplate, getConfig } = require('../utils'
 router.get('/', handlePage)
 router.get('/:page', handlePage)
 
+router.get('/filename-listing.json', async (req, res) => {
+  res.header('Cache-Control', 'public, must-revalidate') // override no-cache
+  const tree = await getTree()
+  res.json({filenames: buildFilenameListing(tree)})
```

Comment:
Hey @afischer, I liked the way we were saving the filenames available during tree generation before. Could we continue that approach just without putting the filenames into the cache layer itself? 

In that case, we'd save the filenames available in scope inside `list.js` the same way we do for `getTree`. We could probably modify `produceTree()` so that it returns an object with `tree` and `filenames` as properties, then destructure the result inside `updateTree()`. A new variable `currentFilenames` could be declared at the top of `list.js` next to `currentTree`, which we override each time we run `updateTree()`. The last step would be adding a companion function to `getTree()`, perhaps called `getFilenames()` that would return the filenames from memory.

What do you think?